logstash-core/lib/logstash/queue_factory.rb

```diff
@@ -24,7 +24,7 @@ def self.create(settings)
         LogStash::Util::WrappedSynchronousQueue.new
       when "persisted"
         # persisted is the disk based acked queue
-        queue_path = settings.get("path.queue")
+        queue_path = ::File.join(settings.get("path.queue"), settings.get("pipeline.id"))
         LogStash::Util::WrappedAckedQueue.create_file_based(queue_path, queue_page_capacity, queue_max_events, checkpoint_max_writes, checkpoint_max_acks, checkpoint_max_interval, queue_max_bytes)
```

Comment:
I agree, I just created this wip PR to serve as discussion "platform" to decide where to put the "create folder" code.
I'll definitely add a test 👍